saleor/userprofile/views.py

```diff
@@ -9,26 +9,50 @@
 from django.template.response import TemplateResponse
 from django.utils.translation import ugettext as _
 
+from ..wishlist.forms import WishlistSettingsForm
+from ..wishlist.models import Wishlist
+
 
 @login_required
 def details(request):
     password_form = get_or_process_password_form(request)
+    wishlist_form = get_or_process_wishlist_form(request)
     ctx = {'addresses': request.user.addresses.all(),
            'orders': request.user.orders.prefetch_related('groups__items'),
-           'change_password_form': password_form}
-
+           'change_password_form': password_form,
+           'wishlist_form': wishlist_form}
     return TemplateResponse(request, 'userprofile/details.html', ctx)
 
 
 def get_or_process_password_form(request):
-    form = ChangePasswordForm(data=request.POST or None, user=request.user)
-    if form.is_valid():
-        form.save()
-        logout_on_password_change(request, form.user)
-        get_adapter(request).add_message(
-            request,
-            messages.SUCCESS,
-            'account/messages/password_changed.txt')
+    if 'change_password_form' in request.POST:
+        form = ChangePasswordForm(data=request.POST, user=request.user,
+                                  prefix='password')
+        if form.is_valid() and request.method == 'POST':
+            form.save()
+            logout_on_password_change(request, form.user)
+            get_adapter(request).add_message(
+                request,
+                messages.SUCCESS,
+                'account/messages/password_changed.txt')
+    else:
+        form = ChangePasswordForm(data=None, user=request.user,
+                                  prefix='password')
+    return form
+
+
+def get_or_process_wishlist_form(request):
+    wishlist, created = Wishlist.objects.get_or_create(user=request.user)
+    if 'wishlist_form' in request.POST:
```

Comment:
You don't have to check this here, when form has a `prefix` set, it won't be validated when there's not enough data in POST.